### PR TITLE
Annotator final polish changes

### PR DESCRIPTION
**Pull request by maciejdrozdowski**
- resolve most TODO fragments from code
- standardize most of common types in Annotator ( preferences -> settings), add ImportObj type

- display metadata in tooltip
- allow to add/edit metadata in annotation Dialog

- clear stash on save
- clear TOOL action on SHAPE unite

- rename CategoryGroup & AnnotationGroup components + adjust props

- add copy annotations action & modal
- add downloadCOCO action
- add missing key actions
- usePolygon - add close action
- add restoreDefaultSettings button in modal
- Standardize textFields in Panels

### CODE REVIEW

File: client/src/Annotator/Info/useDataset.tsx [hunk 1]
```diff
@@ -58,11 +64,29 @@ const useDataset: IUseDataset = imageId => {
         [enqueueSnackbar],
     );
 
+    const copyAnnotations = useCallback(
+        async (imageId: number, id: number, categoriesIds: number[]) => {
```
Comment (wojciech-bilicki) [inline client/src/Annotator/Info/useDataset.tsx]: I would say that function taking 3 args should use named args (pass the object into the function)

File: client/src/Annotator/Info/useInfo.tsx [hunk 2]
```diff
@@ -264,6 +291,46 @@ const useEditor: ISubHook<UseEditorResponse> = (data, setData) => {
...
+
+            const newArr = [...data];
+            newArr[idx].annotations[aIdx].metadata.push({ key: '', value: '' });
+            setData(newArr);
+        },
+        [data, setData],
+    );
+
+    const editAnnotationMetadata = useCallback(
+        (
```
Comment (wojciech-bilicki) [inline client/src/Annotator/Info/useInfo.tsx]: ditto

File: client/src/Annotator/Paper/Component/AnnotationGroupComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
+/**
+ * Fake component which observe changes in info layer
+ * and react for them in a form of changes in paper.js scope object
```
Comment (wojciech-bilicki) [inline client/src/Annotator/Paper/Component/AnnotationGroupComponent.tsx]: `observes` `reacts` `to them`

File: client/src/Annotator/Paper/Component/CategoryGroupComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
+/**
+ * Fake component which observe changes in info layer
+ * and react for them in a form of changes in paper.js scope object
```
Comment (wojciech-bilicki) [inline client/src/Annotator/Paper/Component/CategoryGroupComponent.tsx]: ditto

File: client/src/Annotator/Paper/Shape/KeypointShape.ts [hunk 5]
```diff
@@ -13,7 +13,7 @@ class KeypointShape extends paper.Path {
         // Copy segments from Path.Circle into new paper.Path
         // Keypoint has to inherit from paper.Path
         // because is no such class as paper.Path.Circle... only constructor :-(
```
Comment (wojciech-bilicki) [inline client/src/Annotator/Paper/Shape/KeypointShape.ts]: `because there is no`

File: client/src/Annotator/Paper/Tool/usePolygon.tsx [hunk 6]
```diff
@@ -172,6 +172,8 @@ export const usePolygon: IToolPolygon = (
         cache.current.polygon.removeSegments(length - 1, length);
     }, []);
 
+    const closePath = useCallback(() => _complete(), [_complete])
```
Comment (wojciech-bilicki) [inline client/src/Annotator/Paper/Tool/usePolygon.tsx]: no need to create new function

**Review by wojciech-bilicki — CHANGES_REQUESTED** [review]

**Review by wojciech-bilicki — APPROVED** [review]

**Review by wiola-dreszler — APPROVED** [review]

**Pull request merged**